### PR TITLE
Update and fix permissions logic

### PR DESCRIPTION
**Pull request by wu-clan**
This PR contains significant updates.

### CODE REVIEW

**Comment by wu-clan** [comment]
1: Add role directory authentication

This is not compatible with casbin, enable this method, casbin will be disabled

2：Update base routing

The base route is now `/api/v1`

**Comment by wu-clan** [comment]
please review.

**Comment by downdawn** [comment]
Your current practice is to use the menu to control authority?

My opinion is:

Menu table, permission table, role table, role menu table, role permission table

The menu and permissions are crud normally, but you need to specify the menu id and permission id when creating a role.

**Comment by wu-clan** [comment]
Adding a permission table would be more rigorous, I agree

But if you are using casbin, in fact, the menu permission field has no substantial use, just control the permission by casbin

Using the role menu to control permissions is a classic case of [ruoyi](https://ruoyi.vip/) system, which is a bit simpler

If the permission table is added, the existence of casbin seems meaningless

And the current model looks more flexible, I think so

**Comment by downdawn** [comment]
I got it wrong, there is no associated role when creating a menu, only get_user_menu_tree needs it.

Yes, the authority table is not necessarily required, it is also possible to use casbin

But not all interfaces need DependsRBAC, right?

Interfaces like new_token only need to log in, so you can keep DependsJwtAuth

**Comment by wu-clan** [comment]
We made a mistake, it was wrong to keep only DependsJwtAuth, it just checked that the request header contained the token and did not test if it was real, I re-submitted the code and I think it works now

**Comment by wu-clan** [comment]
Contains a bug in #130, the code field now defaults to string type instead of int or string

**Review by downdawn — APPROVED** [review]

**Comment by downdawn** [comment]
OK

File: backend/app/api/v1/auth/auth.py [hunk 1]
```diff
@@ -42,14 +42,14 @@ async def user_login(request: Request, obj: AuthLogin, background_tasks: Backgro
     return await response_base.success(data=data)
 
 
-@router.post('/new_token', summary='创建新 token', dependencies=[DependsJwtAuth])
+@router.post('/new_token', summary='创建新 token', dependencies=[DependsRBAC])
```
[the remark this answers is not in the document]
Comment (wu-clan) [reply backend/app/api/v1/auth/auth.py]: Yes, I think so

**Comment by downdawn** [comment]
> We made a mistake, it was wrong to keep only DependsJwtAuth, it just checked that the request header contained the token and did not test if it was real, I re-submitted the code and I think it works now

Both DependsJwtAuth and jwt_auth_middleware check whether the header contains Authorization and bearer. If so, jwt_auth_middleware has called jwt_authentication.

So `DependsJwtAuth = Depends(oauth2_schema)` is reasonable

**Comment by wu-clan** [comment]
Agree with you, and RBAC needs to be changed as well, check #132

**Pull request merged**